src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.CitizenDao;
+import core.basesyntax.exceptions.IncorrectAnnotation;
+import core.basesyntax.factory.Factory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+
+    public static Object getInstance(Class<?> clazz) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException, InstantiationException {
+        Constructor<?> constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+        Class<? extends BetDao> betDaoImplClass = Factory.getBetDao().getClass();
+        Class<? extends CitizenDao> citizenDaoClass = Factory.citizenDao().getClass();
+        if (!betDaoImplClass.isAnnotationPresent(Dao.class)
+                || !citizenDaoClass.isAnnotationPresent(Dao.class)) {
+            throw new IncorrectAnnotation("Incorrect annotation!");
```

Comment:
```suggestion
            throw new IncorrectAnnotation("Incorrect annotation!");
```
Let's reconsider the name of this exception